console/v1/types_console_sample.go

```diff
@@ -0,0 +1,183 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsoleSample is an extension to customizing OpenShift web console by adding samples.
+//
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
```

Comment:
Currently `console.openshift.io/v1` is documented as tier 2...  I vaguely recall @bparees suggesting this, but don't remember. @bparees?

https://docs.openshift.com/container-platform/4.13/rest_api/understanding-api-support-tiers.html#mapping-support-tiers-to-openshift-api-groups_understanding-api-tiers

Reply:
I don't remember how we arrived at that value, it seems like that's what it was set to when the docs were first introduced:

https://github.com/openshift/openshift-docs/pull/30286

this looks like the draft doc where tiers were discussed:
https://docs.google.com/document/d/1dkfqbGfi__Cs9UZMZrv1zGaJwXoHxO84YQkMlvrfAo4/edit#heading=h.s3nekardp0u9

but i don't see console explicitly called out there.

That said, as you note, it's documented as an exception, so unless the console team *wants* to change to tier 1, it seems fine that it this new resource would fall under that tier 2 exception.

Reply:
Curious why the console team would want tier 2? We are declaring this a v1 API (where tier 2 is equivalent to a beta API). Is it possible that it was tier 2 because some console stuff was introduced under a v1alpha1 API group?

Realistically, looking at APIs in openshift, while we have declared some tier 2 APIs, we never treat these APIs as tier 2. We only ever treat them as tier 1.

This _should_ be tier 1, and if it isn't ready to go to tier 1, then it should be tier 4, behind a v1alpha1 API group, and behind a feature gate.

Are the console team confident in their assertion that this should be v1 straight away? If they are, I don't know why they wouldn't accept the Tier 1 responsibilities

Reply:
We should update the docs.  Level 2 was created to match up with kubernetes beta APIs.  Since the guide was developed, several things have shifted
1. new kubernetes beta APIs are not enabled by default
2. TechPreview is now usable with new APIs and new fields
3. TechPreview jobs now allow CI to run against them to allow testing before delivery into the product

Level 2 is still reasonable for kubernetes APIs as they arrive, but for openshift specific APIs, the value of a level 2 for both customers and developers is unclear and we should avoid it.

Reply:
> the benefit for developers seems pretty clear: they can remove the api in the future(without waiting for an openshift 5.x) if they choose to do so(over a 3 release schedule), which you can't do with a tier 1 api under our current rules.

Given the conversations I've had with management/PM/staff engineers over the last 2 or 3 years, I don't think there's any appetite to treat Tier 2 APIs as Tier 2. For example the Machine API, all declared beta and tier 2, yet, whenever I've spoken to anyone about this, they say removing it would be OpenShift 5.x. I would expect this and any other Tier 2 declared API to come under that same scrutiny.

I appreciate the sentiment and the supposed value, but I don't think our support requirements/management decisions for OpenShift 4 actually allow tier 2 to exist as it suggests it does.

Reply:
I changed it to tier 1 as we explicitly said that this will be released as GA.